crates/sui-graphql-rpc/src/types/object.rs

```diff
@@ -566,6 +616,155 @@ impl Object {
 
         Ok(conn)
     }
+
+    async fn query_live(db: &Db, address: SuiAddress) -> Result<Option<Self>, Error> {
+        let vec_address = address.into_vec();
+        use objects::dsl as objects;
+
+        let stored_obj: Option<StoredObject> = db
+            .execute(move |conn| {
+                conn.result(move || {
+                    objects::objects
+                        .filter(objects::object_id.eq(vec_address.clone()))
+                        .limit(1)
+                        .into_boxed()
+                })
+                .optional()
+            })
+            .await
+            .map_err(|e| Error::Internal(format!("Failed to fetch object: {e}")))?;
+
+        stored_obj.map(Self::try_from).transpose()
+    }
+
+    async fn query_historical(
+        db: &Db,
+        address: SuiAddress,
+        version: Option<u64>,
+        checkpoint_sequence_number: Option<u64>,
```

Comment:
It seems to me like we either want to do a query by `version` in which case we want to check in the snapshot, or all the historical partitions after the snapshot, or we want to query by `checkpoint_sequence_number`, in which case we want to query for whatever the latest version of that object was at the checkpoint, but I don't know that it makes sense to query with both `version` and `checkpoint_sequence_number`.

Reply:
My original thought was that we should also provide the checkpoint_sequence_number from context (if available), but now that you've called it out, I agree that given a version, it would be an exact match bounded by availableRange, and separately, return the latest version at some checkpoint_sequence_number